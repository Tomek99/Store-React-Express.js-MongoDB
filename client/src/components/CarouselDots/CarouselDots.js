import React from "react";
import styles from "./CarouselDots.module.scss";
import PropTypes from "prop-types";

function CarouselDots({ carouselLength, setSlide, slideIndex }) {
  return (
    <div className={styles.CarouselDots}>
      {Array.from({ length: carouselLength }).map((item, index) => (
        <span
          className={
            index + 1 === slideIndex
              ? `${styles.spanDot} ${styles.activeDot}`
              : styles.spanDot
          }
          key={index}
          onClick={() => setSlide(index + 1)}
        ></span>
      ))}
    </div>
  );
}

CarouselDots.propTypes = {
  length: PropTypes.number,
};

export default CarouselDots;
