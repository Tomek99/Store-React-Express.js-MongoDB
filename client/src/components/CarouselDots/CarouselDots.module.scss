.CarouselDots {
    display: flex;
    justify-content: center;
    gap: 10px;
    position: absolute;
    left: 25px;
    bottom: 20px;
    padding: 10px;
    z-index: 2;
    cursor: default;

    .spanDot {
        width: 7px;
        height: 7px;
        border-radius: 50%;
        background-color: #ccc;
        cursor: pointer;
    }

    .activeDot {
        background-color: var(--main-color);
        border: var(--border);
    }
}
