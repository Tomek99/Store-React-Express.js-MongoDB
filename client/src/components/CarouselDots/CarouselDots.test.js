import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import CarouselDots from "./CarouselDots";
test("Shoulde render CarouselDots component", () => {
  const { container } = render(<CarouselDots carouselLength={3} />);

  expect(container).toMatchSnapshot();
});

test("First dot should be clicked", async () => {
  const mock = jest.fn();
  render(<CarouselDots carouselLength={3} setSlide={mock} />);

  const elements = screen.getAllByTestId("span-element");

  await user.click(elements[0]);

  expect(mock).toHaveBeenCalledWith(1);
});
