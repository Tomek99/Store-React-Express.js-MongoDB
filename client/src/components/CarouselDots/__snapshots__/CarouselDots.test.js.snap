// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoulde render CarouselDots component 1`] = `
<div>
  <div
    class="CarouselDots"
  >
    <span
      class="spanDot"
      data-testid="span-element"
    />
    <span
      class="spanDot"
      data-testid="span-element"
    />
    <span
      class="spanDot"
      data-testid="span-element"
    />
  </div>
</div>
`;
